src/App.tsx

```diff
@@ -1,223 +1,145 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import {
+  FC,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+  useMemo,
+} from 'react';
+import {
+  getTodos,
+  addTodo,
+  deleteTodo,
+} from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
-
-export const App: React.FC = () => {
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Error } from './components/Error/Error';
+import { FilterType } from './components/Enums/FilterType';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+export const App: FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const user = useContext(AuthContext);
+  const user = useContext<User | null>(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filter, setFilter] = useState<FilterType>(FilterType.All);
+  const [isDeletingId, setIsDeletingId] = useState(0);
```

Comment:
currently, when I click the `Clear completed` button and more than 1 todo is completed, only the last todo has a loading circle during deleting
So it should be an array